### PR TITLE
Basic RESTful API with MySQL database and gorp ORM

### PR DESCRIPTION
**Pull request by jamaster**

### CODE REVIEW

File: rooms.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	c.Bind(&room)
+
+	if insert, _ := dbmap.Exec(`INSERT INTO room (Name, Description) VALUES (?, ?)`, room.Name, room.Description); insert != nil {
+		room_id, err := insert.LastInsertId()
+		if err == nil {
+			room.Id = room_id
+			c.JSON(201, room)
+		} else {
+			checkErr(err, "Insert failed")
+		}
```
Comment (poirer) [inline rooms.go]: This will stop the whole application, right? Sounds too-o-o-o-o-o-o hard

File: rooms.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+		v1.GET("/rooms/:id", GetRoom)
+		v1.POST("/rooms", PostRoom)
+		v1.PUT("/rooms/:id", UpdateRoom)
+		v1.DELETE("/rooms/:id", DeleteRoom)
+	}
+
+	r.Run(":8080")
+}
+
+func GetRooms(c *gin.Context) {
```
Comment (poirer) [inline rooms.go]: On my env golint requires to write comments for all functions, variables, constants, etc, which start with upper case. Not sure whose env is wrong...

File: rooms.go (hunk 1 above)
Comment (jamaster) [inline rooms.go]: You are right. I will change error handling.

File: rooms.go [hunk 3]
```diff
@@ -41,96 +45,90 @@ func main() {
...
-	if insert, _ := dbmap.Exec(`INSERT INTO room (Name, Description) VALUES (?, ?)`, room.Name, room.Description); insert != nil {
-		room_id, err := insert.LastInsertId()
-		if err == nil {
+	if insert, err := dbmap.Exec(`INSERT INTO room (Name, Description) VALUES (?, ?)`, room.Name, room.Description); err != nil {
+		fmt.Printf("%+v\n", errors.Wrap(err, "Insert failed"))
+		c.JSON(500, "Room creation failed. Please report us the problem")
+	} else {
+		if room_id, err := insert.LastInsertId(); err != nil {
+			fmt.Print("%+v\n", errors.Wrap(err, "Last inserted id can not be obtained"))
+			c.JSON(500, "Room creation failed. Please report us")
```
Comment (poirer) [inline rooms.go]: Don't know if it is an issue. But "Room created..." is not a JSON. May be you need to use c.String, or c.Text or whatever else than JSON? Won't it fail while trying to present the string as JSON? Won't it add corresponding Content-Type header? If no, then ignore this remark

**Pull request merged**